### PR TITLE
SQS client factory documentation and upgrade guide

### PR DESCRIPTION
**Pull request by danielmarbach**
Doco and upgrade guide for https://github.com/Particular/NServiceBus.AmazonSQS/pull/155

### CODE REVIEW

**Comment by danielmarbach** [comment]
@mauroservienti and or @SimonCropp can you please review?

File: transports/sqs/configuration-options_source_sqs_[4,).partial.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+snippet: S3CredentialSource
+
+## Region
+
+**Mandatory**
+
+**Default**: `AWS SDK`.
+
+By default the endpoint sues the SDK to retrieve the default AWS region from the `AWS_DEFAULT_REGION` environment variable.
```
Comment (mauroservienti) [inline transports/sqs/configuration-options_source_sqs_[4,).partial.md]: typo `sues`

File: samples/sqs/simple/sample.md [hunk 2]
```diff
@@ -21,9 +21,12 @@ Add the [AWS Access Key ID and AWS Secret Access Key](http://docs.aws.amazon.com
 
  * Access Key ID in `AWS_ACCESS_KEY_ID`
  * Secret Access Key in `AWS_SECRET_ACCESS_KEY`
+ * Default Region in `AWS_DEFAULT_REGION`
 
 See also [AWS Account Identifiers](http://docs.aws.amazon.com/general/latest/gr/acct-identifiers.html), [Managing Access Keys for a AWS Account](http://docs.aws.amazon.com/general/latest/gr/managing-aws-access-keys.html), and [IAM Security Credentials](https://console.aws.amazon.com/iam/home#/security_credential).
 
+See also [AWS Regions](http://docs.aws.amazon.com/general/latest/gr/rande.html) for available regions to configure.
```
Comment (mauroservienti) [inline samples/sqs/simple/sample.md]: `for a list of available regions to configure.` or even simpler `for a list of available regions.`

File: Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs [hunk 3]
```diff
@@ -12,4 +17,126 @@ void MaxTTL(EndpointConfiguration endpointConfiguration)
...
+        #endregion
+    }
+
+    void S3CredentialSourceManual(EndpointConfiguration endpointConfiguration, string bucketName, string keyPrefix)
+    {
+        #region 3to4_S3CredentialSourceManual
+
+        var transport = endpointConfiguration.UseTransport<SqsTransport>();
+        var s3Configuration = transport.S3(bucketName, keyPrefix);
+        // SqsCredentialSource.InstanceProfile
```
Comment (mauroservienti) [inline Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs]: What's the purpose of this comment?

File: Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs [hunk 4]
```diff
@@ -12,4 +17,126 @@ void MaxTTL(EndpointConfiguration endpointConfiguration)
...
+        s3Configuration.ClientFactory(() => new AmazonS3Client(new InstanceProfileAWSCredentials()));
+        // or SqsCredentialSource.EnvironmentVariables
```
Comment (mauroservienti) [inline Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs]: What's the purpose of this comment?

File: Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs [hunk 5]
```diff
@@ -12,4 +17,126 @@ void MaxTTL(EndpointConfiguration endpointConfiguration)
...
+        var transport = endpointConfiguration.UseTransport<SqsTransport>();
+
+        #endregion
+    }
+    void CredentialSourceManual(EndpointConfiguration endpointConfiguration)
+    {
+        #region 3to4_CredentialSourceManual
+
+        var transport = endpointConfiguration.UseTransport<SqsTransport>();
+        // SqsCredentialSource.InstanceProfile
```
Comment (mauroservienti) [inline Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs]: What's the purpose of this comment?

File: Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs [hunk 6]
```diff
@@ -12,4 +17,126 @@ void MaxTTL(EndpointConfiguration endpointConfiguration)
...
+        transport.ClientFactory(() => new AmazonSQSClient(new InstanceProfileAWSCredentials()));
+        // SqsCredentialSource.EnvironmentVariables
```
Comment (mauroservienti) [inline Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs]: What's the purpose of this comment?

**Comment by mauroservienti** [comment]
A couple of typos and minor changes, all in all looks very good. I also raised a couple of questions.

File: Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs (hunk 3 above)
Comment (danielmarbach) [reply Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs]: The purpose of this comment is to visualize the previous enum that they had to select. So a mental mapping so to speak

File: Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs (hunk 4 above)
Comment (danielmarbach) [reply Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs]: See https://github.com/Particular/docs.particular.net/pull/3337#discussion_r144469926

File: Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs (hunk 5 above)
Comment (danielmarbach) [reply Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs]: https://github.com/Particular/docs.particular.net/pull/3337#discussion_r144469926

File: Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs (hunk 6 above)
Comment (danielmarbach) [reply Snippets/Sqs/Sqs_4/UpgradeGuides/3to4/Upgrade3to4.cs]: https://github.com/Particular/docs.particular.net/pull/3337#discussion_r144469926

**Comment by danielmarbach** [comment]
@mauroservienti addressed. I still don't have a clue why the engine thinks I have redundant snippets

**Comment by danielmarbach** [comment]
@mauroservienti this is done

**Pull request merged**